media/server/main/source/MediaPipelineServerInternal.cpp

```diff
@@ -587,12 +588,43 @@ bool MediaPipelineServerInternal::haveDataInternal(MediaSourceStatus status, uin
         return true;
     }
     m_activeRequests->erase(needDataRequestId);
+
+    int &counter = m_noAvailableSamplesCounter[mediaSourceType];
+
     if (status != MediaSourceStatus::OK && status != MediaSourceStatus::EOS)
     {
-        RIALTO_SERVER_LOG_WARN("Data request for needDataRequestId: %u received with wrong status", needDataRequestId);
```

Comment:
we still should log this one when there's something wrong and status != MediaSourceStatus::NO_AVAILABLE_SAMPLES